File: test-suite-python/src/test/python/micronaut/tracing/docs/AwsResourceProvider.py
from micronaut.context.annotation import Requires

# tag::imports[]
from io.opentelemetry.contrib.aws.resource import Ec2Resource
from io.opentelemetry.sdk.resources import Resource
from jakarta.inject import Singleton
from micronaut.tracing.opentelemetry import ResourceProvider
# end::imports[]


@Requires(property="spec.name", value="AwsResourceProviderTest")
# tag::clazz[]
@Singleton
class AwsResourceProvider(ResourceProvider):

    def resource(self) -> Resource:
        return Resource.getDefault().merge(Ec2Resource.get())
# end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/tracing/docs/AwsResourceProvider.kt
package io.micronaut.tracing.docs

import io.micronaut.context.annotation.Requires

// tag::imports[]
import io.micronaut.tracing.opentelemetry.ResourceProvider
import io.opentelemetry.contrib.aws.resource.Ec2Resource
import io.opentelemetry.sdk.resources.Resource
import jakarta.inject.Singleton
// end::imports[]

@Requires(property = "spec.name", value = "AwsResourceProviderTest")
// tag::clazz[]
@Singleton
class AwsResourceProvider : ResourceProvider {

    override fun resource(): Resource {
        return Resource.getDefault()
            .merge(Ec2Resource.get())
    }
}
// end::clazz[]
